microsoft/testsuites/kata_conformance/aks_infra.py

```diff
@@ -0,0 +1,136 @@
+
+from azure.identity import DefaultAzureCredential
+from azure.mgmt.containerservice import ContainerServiceClient
+from azure.mgmt.resource import ResourceManagementClient, SubscriptionClient
+
+from lisa.util import LisaException
+from lisa.util.logger import Logger
+
+global_credential_access_lock = Lock()
+
+
+class AKSInfra:
+    def __init__(
+        self,
+        log: Logger,
+        subscription_id: str,
+        tenant_id: str,
+        client_id: str,
+        client_secret: str,
+    ):
+        self.log = log
+        self._credential_setup(subscription_id, tenant_id, client_id, client_secret)
+        self.kube_path = ""
+
+    def _credential_setup(self, subscription_id, tenant_id, client_id, client_secret):
+        if tenant_id:
+            os.environ["AZURE_TENANT_ID"] = tenant_id
+        if client_id:
+            os.environ["AZURE_CLIENT_ID"] = client_id
+        if client_secret:
+            os.environ["AZURE_CLIENT_SECRET"] = client_secret
+        if subscription_id:
+            os.environ["AZURE_SUBSCRIPTION_ID"] = subscription_id
+
+        self._subscription_id = subscription_id
+        self._tenant_id = tenant_id
+        self._client_id = client_id
+        self._client_secret = client_secret
+        self.credential = DefaultAzureCredential()
+        subscription = None
+
+        with SubscriptionClient(self.credential) as self._sub_client:
+            # suppress warning message by search for different credential types
+            with global_credential_access_lock:
+                subscription = self._sub_client.subscriptions.get(self._subscription_id)
+                self.log.info(
+                    f"connected to subscription: "
+                    f"{subscription.id}, '{subscription.display_name}'"
+                )
+
+        if not subscription:
+            raise LisaException(
+                f"Cannot find subscription id: '{self._subscription_id}'. "
+                f"Make sure it exists and current account can access it."
+            )
+
+        self.rsc_mgmt_client = ResourceManagementClient(
+            credential=self.credential, subscription_id=self._subscription_id
+        )
+        self.cntsrv_client = ContainerServiceClient(
+            credential=self.credential, subscription_id=self._subscription_id
+        )
+
+    def create_aks_infra(
+        self, kubernetes_version, worker_vm_size, node_count, azure_region, headers
+    ) -> None:
+        self.log.info("Creating AKS Infra")
+        ts = datetime.datetime.now().strftime("%Y%m%d%H%M%S")
+        aks_cluster_name = f"LISA_AKS_Conf_{ts}"
+        self.resource_group_name = f"LISA_RG_Conf_{ts}"
+        self.log.info("Creating Resource Group : " + self.resource_group_name)
+
+        rg = self.rsc_mgmt_client.resource_groups.create_or_update(
+            self.resource_group_name, {"location": azure_region}
+        )
+        self.log.info("Resource Group Created : " + self.resource_group_name)
```

Comment:
use f string like `f"string: {variable}"`

Reply:
sure, i will use it that way.